### PR TITLE
CIV-15258 fix bug

### PR DESCRIPTION
**Pull request by miguelMolina3691**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-15258

### Change description ###
Fixed bug where the dashboard notification is not removed when the claim is in ‘All final Orders’ state
For the ‘Decision Outcome’ state it has been verified to work correctly

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/main/resources/db/migration/V2024_06_21_1155__CIV-14169-update-claimant-task-list-final-orders-issued.sql [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * Add scenario
```
Comment (ss-evoco) [inline src/main/resources/db/migration/V2024_06_21_1155__CIV-14169-update-claimant-task-list-final-orders-issued.sql:2]: The production database scripts will have been run by the time the code freeze is lifted. All changes to previous scripts need to be done as update SQLs now.

File: src/main/resources/db/migration/V2024_06_21_1166__CIV-14169-update-defendant-task-list-final-orders-issued.sql [hunk 2]
```diff
@@ -2,7 +2,7 @@
  * Add scenario
  */
```
Comment (ss-evoco) [inline src/main/resources/db/migration/V2024_06_21_1166__CIV-14169-update-defendant-task-list-final-orders-issued.sql:3]: The production database scripts will have been run by the time the code freeze is lifted. All changes to previous scripts need to be done as update SQLs now.

**Review by ss-evoco — CHANGES_REQUESTED** [review]

**Review by ss-evoco — DISMISSED** [review]

**Comment by kannan-v-hmcts** [comment]
Hi @miguelMolina3691 The script you have added now will always update the Trial arrangements task list to Inactive when the order is issued. I see 2 issues here.

1. The task list will now appear and be shown as inactive for Small claims as well (Trial arrangements task list is not available for Small claims)
2. For a FT claim, the task list will be always updated to Inactive. I think we should only update it if the trial arrangements is not done. If its done already, we should leave it as Done 

I think we should split the scenario into 2 and update the task list accordingly. Please refer https://github.com/hmcts/civil-service/pull/5463

**Review by dtortolaV1 — APPROVED** [review]

**Review by kannan-v-hmcts — CHANGES_REQUESTED** [review]
Hi @miguelMolina3691 The script you have added now will always update the Trial arrangements task list to Inactive when the order is issued. I see 2 issues here.

The task list will now appear and be shown as inactive for Small claims as well (Trial arrangements task list is not available for Small claims)
For a FT claim, the task list will be always updated to Inactive. I think we should only update it if the trial arrangements is not done. If its done already, we should leave it as Done
I think we should split the scenario into 2 and update the task list accordingly. Please refer https://github.com/hmcts/civil-service/pull/5463

**Review by kannan-v-hmcts — APPROVED** [review]

**Review by laarmada — APPROVED** [review]

**Review by ss-evoco — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/DashboardScenarios.java [hunk 3]
```diff
@@ -195,7 +195,9 @@ public enum DashboardScenarios {
     SCENARIO_AAA6_DISMISS_CASE_CLAIMANT("Scenario.AAA6.CP.Case.Dismissed.Claimant"),
     SCENARIO_AAA6_DISMISS_CASE_DEFENDANT("Scenario.AAA6.CP.Case.Dismissed.Defendant"),
     SCENARIO_AAA6_CP_CASE_STAYED_CLAIMANT("Scenario.AAA6.CP.Case.Stayed.Claimant"),
-    SCENARIO_AAA6_CP_CASE_STAYED_DEFENDANT("Scenario.AAA6.CP.Case.Stayed.Defendant");
+    SCENARIO_AAA6_CP_CASE_STAYED_DEFENDANT("Scenario.AAA6.CP.Case.Stayed.Defendant"),
+    SCENARIO_AAA6_UPDATE_DASHBOARD_DEFENDANT_TASK_LIST_TRIAL_READY_FINAL_ORDERS("Scenario.AAA6.Update.Defendant.TaskList.TrialReady.FinalOrders"),
```
[the remark this answers is not in the document]
Comment (miguelMolina3691) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/DashboardScenarios.java:199]: I have done it this way following the style of the scenario that would be executed if the conditions are not met, which would be this one
SCENARIO_AAA6_UPDATE_DASHBOARD_DEFENDANT_TASK_LIST_UPLOAD_DOCUMENTS_FINAL_ORDERS(‘Scenario.AAA6.Update.Defendant.TaskList.UploadDocuments.FinalOrders’)

What could be a more accurate name? @ss-evoco

**Review by kannan-v-hmcts — DISMISSED** [review]

**Review by ss-evoco — DISMISSED** [review]

**Review by ss-evoco — APPROVED** [review]

**Review by kannan-v-hmcts — APPROVED** [review]

**Pull request merged**